compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -194,8 +193,8 @@ pub struct Elaborator<'context> {
     /// that comptime value and any visibility errors were already reported.
     silence_field_visibility_errors: usize,
 
-    /// Use pedantic ACVM solving
-    pedantic_solving: bool,
+    /// Options from the nargo cli
+    cli_options: ElaboratorOptions<'context>,
```

Comment:
I always think about CLIs as the outer layer about which the core should not know about. It was already a bit strange that `CompileOptions` had `clap` annotations, although useful for defaults, but to see actual `cli_options` and `nargo` referred to in the `noirc_*` modules is strange.

Reply:
Do we really need a separate crate for them though? Could they not be defined next to `CompileOptions`?

Reply:
(By the way the `utils/cli_args` crate seems to be missing from the commits).

Reply:
I spent far too long trying to figure out what this meant only to realize it was never added to git... :man_facepalming:.
Should be added now.

As for the reason why I needed a new crate: it's because `UnstableFeature` is shared between the driver and the frontend so it could only be defined in the frontend since the frontend can't depend on the driver. However this type is also a CLI input and requires clap to derive arguments so I thought it'd be cleaner as a new crate rather than adding clap and CLI functionality to the frontend crate.

Reply:
Sorry, bad phrasing 😄

Right, I see you have the common UnstableFeature type there. I think we could have an UnstableFeature enum for the compiler front end, and other crates would have their own, so conceptually the front end doesn’t need to know about every potentially feature across the repo. And then the CLI can handle the a union and the mappings. 

I share your feelings about clap encroaching on core libraries. I saw people use I think strum to map from strings to enums, which was to support CLIs. In any case depending on a `cli_args` crate is perhaps even more pronounced, but maybe that’s just me.

Reply:
Ok, I managed to reorganize the crates a bit to remove the args-cli crate. We can implement `FromStr` on the type in the frontend and use `value_parser!` on clap in the driver to use that impl and keep them separate.

This took quite a bit longer than I expected though since I was trying to use `FromStr::Error = ()` but clap emits a very cryptic error and it turns out that the unit type is invalid there. I switched it to String and it is now compiling. I'm keeping the same UnstableFeature enum in the CLI but in the future we could create a wrapper enum combining other enums when needed. We'd just need to manually implement ValueParser at that point.